app/models/tenant.rb

```diff
@@ -300,6 +300,7 @@ def validate_only_one_root
   end
 
   def create_tenant_group
+    reload
```

Comment:
Seems slightly neater to keep it all in one transaction to me... and this gives us a sensible place to hang the "see rails/rails#12345" comment